python/cuml/dask/naive_bayes/naive_bayes.py

```diff
@@ -232,7 +235,48 @@ def predict(self, X, delayed=True, parallelism=5):
         dask.Array containing class predictions
         """
 
-        return self._predict(X, delayed, parallelism)
+        gpu_futures = self.client_.sync(extract_arr_partitions, X)
+        x_worker_parts = workers_to_parts(gpu_futures)
+
+        key = uuid1()
+        futures = [(wf[0],
+                    self.client_.submit(MultinomialNB._get_size,
+                                        wf[1],
+                                        workers=[wf[0]],
+                                        key="%s-%s" % (key, idx)))
+                   for idx, wf in enumerate(gpu_futures)]
+
+        sizes = self.client_.compute([x[1] for x in futures], sync=True)
+
+        models = self.client_.scatter(self.local_model,
+                                      broadcast=True,
+                                      direct=True,
+                                      hash=False,
+                                      workers=list(x_worker_parts.keys()))
```

Comment:
Will `scatter` also have issues with models if it has issues with CuPy sparse matrices?

Reply:
No. In fact, the Naive Bayes implementation doesn't actually require serialization of any CuPy sparse arrays and the model itself doesn't store any sparse arrays as state. We need the CuPy sparse array serialization capability so that users can build and manipulate Dask arrays which are backed by CuPy sparse arrays. Currently this is not straightforward, mostly because of a lack of CuPy functionality, but the capability for CuPy sparse arrays to be serializable within the Dask environment is also necessary.